benchmark/single-source/DevirtualizeProtocolComposition.swift

```diff
@@ -0,0 +1,65 @@
+//===--- DevirtualizeProtocolComposition.swift -------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import TestsUtils
+
+public let DevirtualizeProtocolComposition = [
+  BenchmarkInfo(name: "DevirtualizeProtocolComposition", runFunction: run_DevirtualizeProtocolComposition, tags: [.validation, .api]),
+]
+
+public class ClassA<T> { }
+
+protocol ProtocolA {
+  func foo() -> Int
+}
+
+protocol ProtocolB {
+  func bar() -> Int
+}
+
+public class ClassB: ClassA<String> {
+  func foo() -> Int {
+    return 10
+  }
+}
+
+extension ClassB: ProtocolA { }
+
+func quadratic(a: Int) -> Int {
+  var sum = 0
+  for _ in 0..<a {
+    for _ in 0..<a {
+      sum += 1
+    }
+  }
+  return sum
+}
+
+func shouldOptimize1<T>(_ x: ClassA<T> & ProtocolA, count: Int) -> Int {
+  var sum = 0
+  for _ in 0..<count {
+    sum += quadratic(a: x.foo())
+  }
+  return sum
+}
+
+@inline(never)
+public func entryPoint1(c: ClassB) -> Int {
+  return shouldOptimize1(c, count: 25)
```

Comment:
The idea here is that before this patch `shouldOptimize1` counts to 2500 and after this patch, the number 2500 is inlined directly into `entryPoint1` as an `integer_literal`. I may need to add more work to the unoptimized version. 

When I ran the benchmarks on my computer, I thought they were significantly different but, now I'm thinking I may have misread them. The results from running the benchmarks locally:
**Before**:
```
215,DevirtualizeProtocolComposition,1,34914,34914,34914,0,34914
```
**After**:
```
215,DevirtualizeProtocolComposition,100,35,46,36,1,36
```
Anything else I could do to improve this benchmark?

Reply:
Looks like it's too well optimized to measure reliably now. You'll need enough iterations in an "outer" loop such that the measured time is significantly larger than the setup time.

Reply:
Oh, another thing. Please make the non-optimized benchmark faster. We don't want to slow down -Onone benchmarking and it's more than sufficient for one benchmark to slow down by 2x or 10x if this optimization breaks. 10,000x is a bit outrageous.

Reply:
It might help to look at how some of the other devirtualization benchmarks are setup. If you don't know which ones, you could disable the pass and find out. It's nice if the benchmark mimics a realistic scenario, even if it is just a microbenchmark